patients/repo.go

```diff
@@ -1295,70 +1375,208 @@ func PatientsToTideResult(patientsList []*Patient, period string, exclusions *[]
+			{
+				SummaryField:                "timeInAnyLowPercent",
+				SummaryFieldComparator:      "$not",
+				ComparatorOperandExpression: bson.M{"$gt": .04},
+			},
+			{
+				SummaryField:                "timeInVeryLowPercent",
+				SummaryFieldComparator:      "$not",
+				ComparatorOperandExpression: bson.M{"$gt": .01},
+			},
+			{
+				SummaryField:                "timeInHighPercent",
+				SummaryFieldComparator:      "$not",
+				ComparatorOperandExpression: bson.M{"$gt": .25},
+			},
+			{
+				SummaryField:                "timeInVeryHighPercent",
+				SummaryFieldComparator:      "$not",
+				ComparatorOperandExpression: bson.M{"$gt": .05},
+			},
+			{
+				SummaryField:                "timeInExtremeHighPercent",
+				SummaryFieldComparator:      "$not",
+				ComparatorOperandExpression: bson.M{"$gt": .01},
+			},
+		},
+	},
+	{
+		CategoryName:          "timeInExtremeHighPercent",
+		SummaryField:          "timeInExtremeHighPercent",
+		SummaryFieldSortOrder: -1,
+		SummaryFieldFilters: []summaryFieldFilter{
+			{
+				SummaryField:                "timeInExtremeHighPercent",
+				SummaryFieldComparator:      "$gt",
+				ComparatorOperandExpression: 0.01,
+			}},
+	},
+	{
+		CategoryName:          "timeInVeryHighPercent",
+		SummaryField:          "timeInVeryHighPercent",
+		SummaryFieldSortOrder: -1,
+		SummaryFieldFilters: []summaryFieldFilter{
+			{
+				SummaryField:                "timeInVeryHighPercent",
+				SummaryFieldComparator:      "$gt",
+				ComparatorOperandExpression: 0.05,
+			}},
+	},
+	{
+		CategoryName:          "timeInHighPercent",
+		SummaryField:          "timeInHighPercent",
+		SummaryFieldSortOrder: -1,
+		SummaryFieldFilters: []summaryFieldFilter{
+			{
+				SummaryField:                "timeInHighPercent",
+				SummaryFieldComparator:      "$gt",
+				ComparatorOperandExpression: 0.25,
+			}},
+	},
+}
+
+// defaultOrderedCategoryNames is the ORDERED listing of categories by name to
+// use if TideReportParams.Categories is empty. It is ordered because some
+// categories have a higher priority in terms of whether patients within that
+// category should be shown or not due to the restriction on returned results
+// by TideReportPatientLimit
+var defaultOrderedCategoryNames = []string{
+	"timeInVeryLowPercent",
+	"timeInAnyLowPercent",
+	"dropInTimeInTargetPercent",
+	"timeInTargetPercent",
+	"timeCGMUsePercent",
+	"meetingTargets",
+}
+
+// getCategoriesByNames returns a slice of categories with a CategoryName in names. The order of the returned slice matches the order of the name in names.
+func getCategoriesByNames(names []string) []tideCategory {
+	cats := make([]tideCategory, 0, len(names))
+	alreadySeen := make(map[string]bool, len(names))
+	for _, name := range names {
+		if !alreadySeen[name] {
+			alreadySeen[name] = true
+			idx := slices.IndexFunc(availableCategories[:], func(cat tideCategory) bool {
+				return cat.CategoryName == name
+			})
+			if idx > -1 {
```

Comment:
I assume you're ok with silently ignoring a provided name that doesn't match a category? That's something that could come back to bite us later.